internal/integrations/config/loader.go

```diff
@@ -0,0 +1,163 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-viper/mapstructure/v2"
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/v2"
+	"github.com/samber/lo"
+
+	"github.com/theopenlane/core/internal/integrations"
+	"github.com/theopenlane/core/internal/integrations/types"
+)
+
+// FSLoader reads provider specs from an fs.FS rooted at the configured path
+type FSLoader struct {
+	// FS is the filesystem used to read provider specs
+	FS fs.FS
+	// Path is the relative directory containing provider files
+	Path string
+}
+
+// NewFSLoader builds a loader using the supplied filesystem and relative path
+func NewFSLoader(fsys fs.FS, path string) *FSLoader {
+	return &FSLoader{
+		FS:   fsys,
+		Path: path,
+	}
+}
+
+// Load walks the configured directory and decodes every JSON provider file
+func (l *FSLoader) Load(_ context.Context) (map[types.ProviderType]ProviderSpec, error) {
```

Comment:
Doesn't seem like this is needed as input?
```suggestion
func (l *FSLoader) Load() (map[types.ProviderType]ProviderSpec, error) {
```